app/src/main/java/org/jellyfin/androidtv/ui/itemhandling/ItemRowAdapter.java

```diff
@@ -835,14 +843,28 @@ public void addToParentIfResultsReceived() {
     }
 
     private void retrieve(final ItemQuery query) {
+        retrieve(query, false);
+    }
+
+    private void retrieve(final ItemQuery query, final boolean filterPlayed) {
         apiClient.getValue().GetItemsAsync(query, new Response<ItemsResult>() {
             @Override
             public void onResponse(ItemsResult response) {
                 if (response.getItems() != null && response.getItems().length > 0) {
-                    setTotalItems(query.getEnableTotalRecordCount() ? response.getTotalRecordCount() : response.getItems().length);
+                    int totalItems = query.getEnableTotalRecordCount() ? response.getTotalRecordCount() : response.getItems().length;
+                    BaseItemDto[] items = response.getItems();
+                    if (filterPlayed) {
+                      items = Arrays.stream(items).filter(x ->
+                                (x.getUserData() == null) || (!x.getUserData().getPlayed() &&
+                                    x.getUserData().getUnplayedItemCount() != null &&
+                                    x.getUserData().getUnplayedItemCount() > 0)
+                        ).toArray(BaseItemDto[]::new);
```

Comment:
We shouldn't do client side filtering because this is not efficient at all for large libraries.

Reply:
Sure, but this is just a fix i can do now on the client, which was easy to implement.
I don't have the time to also get into the server and working on fixes there.
So until "someone" pick's-up the issue for the server, i rather have working filters for users.